net.veloren.airshipper.yaml

```diff
@@ -25,7 +25,14 @@ modules:
         "CARGO_HOME" : "/run/build/airshipper/cargo",
         "RUST_BACKTRACE" : "1"
       }
-    build-commands : 
+    build-commands :
+      - ls -la /usr/lib/sdk/
+      - ls -la /usr/lib/sdk/rust-nightly
+      - apt-get update
+      - apt-get install -y rustup
+      - rustup toolchain list
```

Comment:
This won't work, it's not a regular CI system that you can install dependencies like a distro.

If you want to add rustup, you should build it from source https://rust-lang.github.io/rustup/installation/other.html#self-compiled-installation and add it to the manifest, or use one of the precompiled binary packages.

Reply:
@bbhtt thanks for the help.

I am struggleing with the fact that ne need an EXACT version of the cargo compiler: https://github.com/flathub/org.freedesktop.Sdk.Extension.rust-nightly/blob/branch/23.08/org.freedesktop.Sdk.Extension.rust-nightly.appdata.xml#L90

however I have no Idea how to import it that way. It seems like whatever I do its always the same version, provided by flathub. that is ... changing daily.

Reply:
> however I have no Idea how to import it that way. It seems like whatever I do its always the same version, provided by flathub. that is ... changing daily.

Yes the nightly rust extension updates daily to the latest version and an app gets built against whichever version is the last published.

If you don't need rust nightly, you should switch to the rust stable extension which follows stable releases of rust.

If you need rust nightly but a specific version of it, then you have to add it in the manifest here and not use the extension. Something like this:

```yaml
  - name: rust
    buildsystem: simple
    sources:
      - type: archive
        only-arches:
          - aarch64
        dest: rust
        # replace version and sha256 here
        url: https://static.rust-lang.org/dist/2024-03-16/rust-nightly-aarch64-unknown-linux-gnu.tar.xz
        sha256: 482c419a2df7bdd2a9eaba20b4aec7bb9990792bb379ec3c92e0d865c807a37c
        x-checker-data:
          type: rust
          package: rust
          channel: nightly
          target: aarch64-unknown-linux-gnu
      - type: archive
        only-arches:
          - x86_64
        dest: rust
        # replace version and sha256 here
        url: https://static.rust-lang.org/dist/2024-03-16/rust-nightly-x86_64-unknown-linux-gnu.tar.xz 
        sha256: 713142167de036156ce7b0a6b80c847e54e31c7a7c51ed222183c3cc669a780a
        x-checker-data:
          type: rust
          package: rust
          channel: nightly
          target: x86_64-unknown-linux-gnu
    build-commands:
      - cd rust && ./install.sh --prefix=/app/sdk/rust-nightly --without=rust-docs
        --without=rust-docs-json-preview --disable-ldconfig --verbose
```

Then when you build the app, append that location to `PATH` `/app/sdk/rust-nightly/bin` so that it can find the executable:

```yaml
build-options:
  append-path: /app/sdk/rust-nightly/bin
```

Finally once the compilation is done, if you no longer need the build-dependencies, you want to cleanup them, to keep the package size in check.

```yaml
cleanup: 
  - /sdk
```

in the top, before `command`